src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -94,78 +94,105 @@ export function DiscoverMainRoute() {
+      );
 
-        if (!currentSavedSearch.searchSource.getField('index')) {
-          currentSavedSearch.searchSource.setField('index', loadedIndexPattern);
-        }
+      if (!loadedIndexPattern) {
+        return;
+      }
 
-        setSavedSearch(currentSavedSearch);
+      if (!currentSavedSearch.searchSource.getField('index')) {
+        currentSavedSearch.searchSource.setField('index', loadedIndexPattern);
+      }
 
-        if (currentSavedSearch.id) {
-          chrome.recentlyAccessed.add(
-            getSavedSearchFullPathUrl(currentSavedSearch.id),
-            currentSavedSearch.title ?? '',
-            currentSavedSearch.id
-          );
-        }
-      } catch (e) {
-        if (e instanceof DataViewSavedObjectConflictError) {
-          setError(e);
-        } else {
-          redirectWhenMissing({
-            history,
-            navigateToApp: core.application.navigateToApp,
-            basePath,
-            mapping: {
-              search: '/',
-              'index-pattern': {
-                app: 'management',
-                path: `kibana/objects/savedSearches/${id}`,
-              },
-            },
-            toastNotifications,
-            onBeforeRedirect() {
-              getUrlTracker().setTrackedUrl('/');
+      setSavedSearch(currentSavedSearch);
+
+      if (currentSavedSearch.id) {
+        chrome.recentlyAccessed.add(
+          getSavedSearchFullPathUrl(currentSavedSearch.id),
+          currentSavedSearch.title ?? '',
+          currentSavedSearch.id
+        );
+      }
+    } catch (e) {
+      if (e instanceof DataViewSavedObjectConflictError) {
+        setError(e);
+      } else {
+        redirectWhenMissing({
+          history,
+          navigateToApp: core.application.navigateToApp,
+          basePath,
+          mapping: {
+            search: '/',
+            'index-pattern': {
+              app: 'management',
+              path: `kibana/objects/savedSearches/${id}`,
             },
-            theme: core.theme,
-          })(e);
-        }
+          },
+          toastNotifications,
+          onBeforeRedirect() {
+            getUrlTracker().setTrackedUrl('/');
+          },
+          theme: core.theme,
+        })(e);
       }
     }
-
-    loadSavedSearch();
   }, [
+    id,
+    services.data.search,
+    services.spaces,
     core.savedObjects.client,
-    basePath,
-    chrome.recentlyAccessed,
-    config,
     core.application.navigateToApp,
-    data.dataViews,
+    core.theme,
+    loadDefaultOrCurrentIndexPattern,
+    chrome.recentlyAccessed,
     history,
-    id,
-    services,
+    basePath,
     toastNotifications,
-    core.theme,
-    checkForDataViews,
   ]);
 
+  const onDataViewCreated = useCallback(
+    async (dataView: unknown) => {
+      const dataViewId = (dataView as DataView).id;
+      if (dataViewId) {
+        setShowNoDataPage(false);
+        setError(undefined);
+        await loadSavedSearch();
+      }
```

Comment:
yes, the typecast is here because the method will be called with an "unknown" parameter. as for the id - I don't think so? as soon as the data view is created, it will have an id.

Reply:
Removed the `dataViewId` as we don't really need it.

Reply:
Great, not that now `unknown` of `async (dataView: unknown)` is redundant